### ISSUE TITLE
[Edited] No CUDA runtime is found, using CUDA_HOME='C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8'

### ISSUE THREAD

**Issue by yoponchik** [graphdeco-inria/gaussian-splatting#748]
```
~
Failed to build diff_gaussian_rasterization simple_knn
Installing collected packages: simple_knn, diff_gaussian_rasterization
  Running setup.py install for simple_knn: started
  Running setup.py install for simple_knn: finished with status 'error'

Pip subprocess error:
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      running bdist_wheel
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:411: UserWarning:.
        warnings.warn(msg.format('we could not find ninja.'))
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-37
      creating build\lib.win-amd64-cpython-37\diff_gaussian_rasterization
      copying diff_gaussian_rasterization\__init__.py -> build\lib.win-amd64-cpython-37\diff_gaussian_rasterization
      running build_ext
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:813: UserWarning:.
        warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
      building 'diff_gaussian_rasterization._C' extension
      creating build\temp.win-amd64-cpython-37
      creating build\temp.win-amd64-cpython-37\Release
      creating build\temp.win-amd64-cpython-37\Release\cuda_rasterizer
      "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\bin\nvcc" -c cuda_rasterizer/backward.cu -o build\temp.v
      C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.8\include\crt/host_config.h(231): fatal error C1083: Cannoy
      backward.cu
      error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.8\\bin\\nvcc.exe' failed with exit cod2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for diff_gaussian_rasterization
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [12 lines of output]
      running bdist_wheel
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:411: UserWarning:.
        warnings.warn(msg.format('we could not find ninja.'))
      running build
      running build_ext
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:813: UserWarning:.
        warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
      building 'simple_knn._C' extension
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\bin\HostX64\x64\cl.exe" /c /no0
      ext.cpp
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\include\torch\csrc\api\include\torch/cudy
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.33.31629\\bin\\H2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for simple_knn
  error: subprocess-exited-with-error

  × Running setup.py install for simple_knn did not run successfully.
  │ exit code: 1
  ╰─> [28 lines of output]
      running install
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\setuptools\_distutils\cmd.py:66: SetuptoolsDep.
      !!

              ********************************************************************************
              Please avoid running ``setup.py`` directly.
              Instead, use pypa/build, pypa/installer or other
              standards-based tools.

              See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
              ********************************************************************************

      !!
        self.initialize_options()
      running build
      running build_ext
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:411: UserWarning:.
        warnings.warn(msg.format('we could not find ninja.'))
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\utils\cpp_extension.py:813: UserWarning:.
        warnings.warn(CUDA_MISMATCH_WARN.format(cuda_str_version, torch.version.cuda))
      building 'simple_knn._C' extension
      creating build
      creating build\temp.win-amd64-cpython-37
      creating build\temp.win-amd64-cpython-37\Release
      "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629\bin\HostX64\x64\cl.exe" /c /no0
      ext.cpp
      C:\Users\ITAEC\anaconda3\envs\gaussian_splatting\lib\site-packages\torch\include\torch\csrc\api\include\torch/cudy
      error: command 'C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.33.31629\\bin\\H2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> simple_knn

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

failed
~
```

**Comment by Muhammad0312** [comment]
Hello,
Were you able to solve this issue?

**Comment by zniihgnexy** [comment]
@yoponchik Hi thank you so much for your environment file provided above. That works for me. I'd like to ask about the submodules. What should i do after i downloaded them? Did you build then by running setup.py file since this didn't work for me and i'm confused about how can i use these two folders.

**Comment by YuehuGong** [comment]
I also encountered this prolblem in my ubuntu and solved it. And I'd like to introduce the confusing way that I solved it.
Firstly, I updated my gcc and g++ from 9 to 11, and set 11  as default with update-alternatives.
Then I set the $CUDA_HOME variable to /usr/local/cuda-11.8. and it worked.
**Attention:** Originally I use export to temporarily set the CUDA_HOME variable but it was completely useless. Then I use `echo 'export CUDA_HOME=/usr/local/cuda-11.8' >> ~/.bashrc` command to set the variable permanently and it worked.

**Comment by YuehuGong** [comment]
I encounter the same problem in ubuntu and python 3.9.
It's fixed when I use a new python 3.8 environment.

**Comment by byte118** [comment]
**For LINUX OS. I figure out this problem!** It's all because the "environment.yml" which is default to install "pytorch" for cpu version. So i met "No CUDA runtime is found, using CUDA_HOME='/usr/local/cuda-11.8" problem. **TO SOLVE THIS,**  i first uninstall the pytorch. And reinstall pytorch for linux version by "**conda install pytorch==2.0.0 torchvision==0.15.0 torchaudio==2.0.0 pytorch-cuda=11.8 -c pytorch -c nvidia**"
